tck/src/main/java/org/eclipse/microprofile/opentracing/tck/OpenTracingBaseTests.java

```diff
@@ -227,25 +228,46 @@ protected void assertEqualTrees(ConsumableTree<TestSpan> returnedTree,
         // It's okay if the returnedTree has tags other than the ones we
         // want to compare, so just remove those
         returnedTree.visitTree(span -> {
-            if (Tags.SPAN_KIND_SERVER.equals(span.getTags().get(Tags.SPAN_KIND.getKey()))) {
-                span.getTags().keySet()
-                    .removeIf(key -> !key.equals(Tags.SPAN_KIND.getKey())
-                        && !key.equals(Tags.HTTP_METHOD.getKey())
-                        && !key.equals(Tags.HTTP_URL.getKey())
-                        && !key.equals(Tags.HTTP_STATUS.getKey())
-                        && !key.equals(Tags.COMPONENT.getKey())
-                        && !key.equals(TestServerWebServices.LOCAL_SPAN_TAG_KEY));
-            }
-            else {
-                span.getTags().keySet()
-                    .removeIf(key -> !key.equals(Tags.SPAN_KIND.getKey())
-                        && !key.equals(Tags.HTTP_METHOD.getKey())
-                        && !key.equals(Tags.HTTP_URL.getKey())
-                        && !key.equals(Tags.HTTP_STATUS.getKey())
-                        && !key.equals(Tags.COMPONENT.getKey())
-                        && !key.equals(TestServerWebServices.LOCAL_SPAN_TAG_KEY)
-                        && !key.equals(Tags.ERROR.getKey()));
-            }
+            span.getTags().keySet()
+                .removeIf(key -> !key.equals(Tags.SPAN_KIND.getKey())
+                    && !key.equals(Tags.HTTP_METHOD.getKey())
+                    && !key.equals(Tags.HTTP_URL.getKey())
+                    && !key.equals(Tags.HTTP_STATUS.getKey())
+                    && !key.equals(Tags.COMPONENT.getKey())
+                    && !key.equals(TestServerWebServices.LOCAL_SPAN_TAG_KEY));
+        });
+
+        // It's okay if the returnedTree has log entries other than the ones we
+        // want to compare, so just remove those
+        returnedTree.visitTree(span -> span.getLogEntries()
```

Comment:
If I understand it correctly the only significant change in this PR is this addition. 

You split the method but also changed client tests to call new method `assertEqualErrorTrees`. If the split is not needed I would prefer to keep it in one method.

Reply:
The current way is to use to `Tags.SPAN_KIND` to determine to remove the `error`, `event` and `error.obejct` from `logEntries does not work for all cases.  It is more flexible to split them or we need to find another condition to choose which section to run in a combined method.